src/main/java/CheonsaBot.java

```diff
@@ -0,0 +1,421 @@
+                        throw new IllegalArgumentException("Oops! Missing description for 'event' :( ");
+                    }
+                    break;
+
+                default:
+                    throw new IllegalArgumentException("Oh no! Unknown task type: " + command + " :(\nTry commands todo, deadline, or event with the correct format!");
+            }
+
+            if (task != null) {
+                tasks.add(task);
+                appendTaskToFile(task); // Append the task to the file
+                System.out.println(getHorizontalLine());
+                System.out.println("Added: " + task);
+                System.out.println(getHorizontalLine());
+            }
+
+        } catch (IllegalArgumentException e) {
+            System.out.println(getHorizontalLine());
+            System.out.println("Error: " + e.getMessage());
+            System.out.println(getHorizontalLine());
+        }
+    }
+  
+    /*
+     * Delete the task at specified index
+     * @param index The index of task to delete.
+     */
+    private static void deleteTask(String index) {
+        try {
+            int taskIndex = Integer.parseInt(index) - 1;
+            if (taskIndex < 0 || taskIndex >= tasks.size()) {
+                throw new IndexOutOfBoundsException();
+            }
+    
+            System.out.println(getHorizontalLine());
+            System.out.println("Removed: " + tasks.get(taskIndex).getDescription());
+            tasks.remove(taskIndex);
+            updateTaskFile(); // Rewrite the file after deletion
+            System.out.println("Only " + tasks.size() + " thing(s) left to do!");
+            System.out.println(getHorizontalLine());
+    
+        } catch (NumberFormatException | IndexOutOfBoundsException e) {
+            System.out.println(getHorizontalLine());
+            System.out.println("Error: Invalid task number! :(");
+            System.out.println(getHorizontalLine());
+        }
+    }
+
+    /**
+     * Appends a task to the tasklist.txt file.
+     * @param task The task to write to the file on disk.
+     */
+    private static void appendTaskToFile(Task task) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(FILE_PATH, true))) {
+            writer.write(formatTaskForFile(task));
+            writer.newLine();
+        } catch (IOException e) {
+            System.out.println("Oh no, error appending task to file: " + e.getMessage() + " :(");
+        }
+    }
+
+    /**
+     * Formats a task as a string for saving to the file.
+     * @param task The task to write to the file on disk.
+     * @return A line to be saved to file on disk.
+     */
+    private static String formatTaskForFile(Task task) {
+        String taskType = "";
+        if (task instanceof ToDo) {
+            taskType = "T";
+        } else if (task instanceof Deadline) {
+            taskType = "D";
+        } else if (task instanceof Event) {
+            taskType = "E";
```

Comment:
You may consider converting task types (i.e. "T", "D", "E") into named constants. This also applies to other magic literals especially when they are used multiple times in the code.